### ISSUE TITLE
Dev service worker on Firefox is not working

### ISSUE THREAD

**Issue by fkrauthan** [vite-pwa/sveltekit#95]
I've configured the service worker to use `strategies: "injectManifest",` and created the minimal service worker code required. My dev section looks as follow:

```
			devOptions: {
				enabled: true,
				type: "module",
			},
```

However when running my sveltekit app in dev mode (`pnpm run dev`) and opening the page on Firefox I get a `SW registration error TypeError: ServiceWorker script at http://localhost:5173/dev-sw.js?dev-sw for scope http://localhost:5173/ threw an exception during script evaluation. [PWA.svelte:23:18](http://localhost:5173/src/lib/components/PWA.svelte)` error (without any additonal errors displayed in the console).

Opening the same page in for example Chrome registers the service worker correctly.

You can use this repository https://github.com/fkrauthan/sveltekit-pwa-issue to see this issue yourself.

**Comment by userquin** [comment]
You cannot use custom sw with type module in Firefox, it is not supported.

**Comment by fkrauthan** [comment]
Hmm but changing `type` to `classic` produces the same error?

**Comment by userquin** [comment]
Because cannot use `import` statements, the custom sw in dev is "compiled" via esbuild just removing the ts annotations:
- https://caniuse.com/?search=service%20worker
- https://caniuse.com/mdn-javascript_statements_import_service_worker_support

Check the source code in the network or source tab.

**Comment by fkrauthan** [comment]
Ah so in general dev mode for service workers does not work right now. Might be worth pointing out in the docs (as it only calls out that `type` `module` is only supported on chromium.

**Comment by userquin** [comment]
You can write your custom service worker using `importScripts`, it should work.

**Comment by userquin** [comment]
> Ah so in general dev mode for service workers does not work right now for firefox. Might be worth pointing out in the docs (as it only calls out that `type` `module` is only supported on chromium.

https://vite-pwa-org.netlify.app/guide/development.html#injectmanifest-strategy

**Comment by fkrauthan** [comment]
Yeah to me that reads as I can use `type` `module` on chrome (especially since the box just below it talks about that `classic` is used for prod builds all the time and works on all browsers).
